inspirehep/modules/records/texkeys.py

```diff
@@ -0,0 +1,229 @@
+    new_texkey = ''
+    first_part = first_part_generator(record)
+    if first_part:
+        new_texkey = first_part + ':' + second_part_generator(record) + third_part_generator()
+    return new_texkey
+
+
+def first_part_generator(record):
+    if len(record.get('authors', [])) in range(1, 10):
+        first_part = sanitize(record.get('authors')[0]['full_name'])
+    elif len(record.get('collaborations', [])):
+        first_part = sanitize(record.get('collaborations')[0]['value'])
+    elif len(record.get('corporate_author', [])):
+        first_part = sanitize(record.get('corporate_author')[0])
+    elif 'proceedings' in record.get('document_type', []):
+        first_part = 'proceedings'
+    elif len(record.get('authors', [])):
+        first_part = sanitize(record.get('authors')[0]['full_name'])
+    else:
+        first_part = ''
+
+    return first_part
+
+
+def second_part_generator(record):
+    return get_year_from_record(record)
+
+
+def get_three_random_letters():
+    letters = string.letters.lower()
+    random_sequence = ''
+    for _ in range(3):
+        random_sequence += random.choice(letters)
+
+    return random_sequence
+
+
+def third_part_generator():
+    random_sequence = get_three_random_letters()
+    return random_sequence
+
+
+def is_current_texkey_valid(record):
+    fixed_part = first_part_generator(record) + ':' + second_part_generator(record)
+    previous_texkeys = record.get('texkeys')
+    if previous_texkeys:
+        return fixed_part in previous_texkeys[0]
+    return False
+
+
+def store_texkeys_pidstore(new_texkey, record, obj_uuid):
+    """It stores the tekeys in the pidstore
+    table if has not been used before.
+
+    :param new_texkey: the texkey that has to be saved
+    :type new_texkey: string
+
+    :param record: the record related to the texkey
+    :type new_texkey: string
+
+    :return: return False if the key is already present,
+    True if the key has been stored correctly
+    :rtype: boolean
+    """
+    try:
+        PersistentIdentifier.get('tex', new_texkey)
+        return False
+    except PIDDoesNotExistError:
+        TexKeyIdProvider.create(
+            'tex',
+            obj_uuid,
+            pid_value=new_texkey
+        )
+        db.session.commit()
+    return True
+
+
+def sanitize(value):
+    def _remove_unicode_special_chars(value):
+        return re.sub(
+            r'[^-A-Za-z0-9.:/^_;&*<>?|!$+]',
+            '',
+            value
+        )
+
+    def _are_all_letters(value):
+        return re.search(
+            r'[A-Za-z]',
+            value
+        )
+
+    # Remove utf-8 special characters
+    value = unidecode(
+        value.encode('utf-8').decode('utf-8')
+    )
+    # Take the last name
+    try:
+        value = value.split(',')[0]
+    except KeyError:
+        value = ''
+
+    value = _remove_unicode_special_chars(value)
+    if len(value) == 0 or not _are_all_letters(value):
+        value = ''
+
+    return value
```

Comment:
You might be reimplementing [`asciify`](https://github.com/inspirehep/beard/blob/666e3b8f0449af4818f9903fd1f712733ce8ca73/beard/utils/strings.py#L27-L37).